Support pytest-bdd 5

## Bug Report
<!--
Hello, and thank you for reporting an issue!
Please fill out the points below, as it will make our process much easier.
-->

<!-- Please tell us about your environment -->
### Environment

I’m the maintainer of the `jrnl` and `python-pytest-bdd` packages in Fedora Linux. This report is independent of environment.

### Current Behavior
<!--
Please put a short description of what is currently happening.
-->

Version 5.0.0 of `pytest-bdd` was released today. Some small adjustments will be required to support it; upstream has published a [migration guide](https://github.com/pytest-dev/pytest-bdd/tree/5.0.0#migration-of-your-tests-from-versions-4xx).

Currently, if I run the tests with `pytest-bdd` 5.x, I see only a few tests that look like they need to be adjusted:

```
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_one[Monday: entry on a monday-entry on a monday-2019-03-11 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_one[Tuesday: entry on a tuesday-entry on a tuesday-2019-03-05 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_one[Wednesday: entry on a wednesday-entry on a wednesday-2019-03-06 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_one[Thursday: entry on a thursday-entry on a thursday-2019-03-07 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_one[Friday: entry on a friday-entry on a friday-2019-03-08 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_one[Saturday: entry on a saturday-entry on a saturday-2019-03-09 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_one[Sunday: entry on a sunday-entry on a sunday-2019-03-10 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_one[sunday: entry on a sunday-entry on a sunday-2019-03-10 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_one[sUndAy: entry on a sunday-entry on a sunday-2019-03-10 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_two[Mon: entry on a monday-entry on a monday-2019-03-11 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_two[Tue: entry on a tuesday-entry on a tuesday-2019-03-05 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_two[Wed: entry on a wednesday-entry on a wednesday-2019-03-06 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_two[Thu: entry on a thursday-entry on a thursday-2019-03-07 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_two[Fri: entry on a friday-entry on a friday-2019-03-08 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_two[Sat: entry on a saturday-entry on a saturday-2019-03-09 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_two[Sun: entry on a sunday-entry on a sunday-2019-03-10 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_two[sun: entry on a sunday-entry on a sunday-2019-03-10 09:00]
FAILED tests/bdd/test_features.py::test_create_entry_using_day_of_the_week_as_entry_date_two[sUn: entry on a sunday-entry on a sunday-2019-03-10 09:00]
```

The `jrnl` tests require `pytest-bdd` 4.x—which is fine!—but eventual support for either `pytest-bdd` 5.x only or for either major version would be helpful.

### Expected Behavior
<!--
Please write a short description of what you would expect to happen
(instead of what is currently happening).
-->

The `jrnl` tests support `pytest-bdd` 5.x (perhaps in addition to 4.x).

### Repro Steps
<!--
Please provide the steps to reproduce the problem. It is important for you to
be as precise as possible here, since more info will let us help you faster.
If applicable, please run jrnl with --debug and paste the output.
-->
Using whatever your usual workflow is, allow/require a newer version of `pytest-bdd`, and try running the tests.

### Other Information
<!-- (e.g. more detailed explanation, stacktraces, related
issues, suggestions how to fix, links for us to have context, eg.
stackoverflow, gitter, etc) -->

I would like to update `python-pytest-bdd` to 5.x in the development version of Fedora Linux without providing a compatibility package for 4.x. Since I control both `python-pytest-bdd` and `jrnl`, and no other packages in the distribution currently use `python-pytest-bdd`, I am not in a hurry. I can wait a while for `jrnl` to support the new version.

When I have a little time to work on it, I’ll look at offering a PR, if nobody else has done it first.

--- a/tests/lib/then_steps.py
+++ b/tests/lib/then_steps.py
@@ -74,6 +74,7 @@ def output_should_be_empty(cli_run):
     assert actual == ""
 
 
+@then(parse('the output should contain the date "{date}"'))
 @then('the output should contain the date "<date>"')
 def output_should_contain_date(date, cli_run):
     assert date and date in cli_run["stdout"]

